### PR TITLE
No directory search for you

### PR DESCRIPTION
**Pull request by vertein**
Allows for configurable directory searches

No Directory example where section just drops out - 
systemnodirectory

Someone in the 'UW-Madison' Group via the configuration
madisondirectory

### CODE REVIEW

File: angularjs-portal-home/src/main/webapp/js/app-config.js [hunk 1]
```diff
@@ -92,6 +91,12 @@ define(['angular'], function(angular) {
             "title" : "Old MyUW"
           },
         ])
+        .value('SEARCH_URLS', [
```
Comment (timlevett) [inline angularjs-portal-home/src/main/webapp/js/app-config.js]: Maybe name this `DIRECTORY_SEARCH_CONFIG` or something like that?

Could also change the structure to be 

```json
[
 {
  "group" : "UW-Madison",
  "url" : "/web/api/wiscdirectory"
 }
]
```

This way we could possibly extend a directory search to other campuses.

File: angularjs-portal-home/src/main/webapp/my-app/search/services.js [hunk 2]
```diff
@@ -33,30 +33,97 @@ define(['angular', 'jquery'], function(angular, $) {
       
     }]);
     
-    app.factory('wiscDirectorySearchService', ['$http', 'miscService', 'SERVICE_LOC', function($http, miscService, SERVICE_LOC){
+    app.factory('directorySearchService', [
+         '$http', '$sessionStorage', '$q', 'PortalGroupService', 'SEARCH_URLS', 'filterFilter', 'miscService', 'SERVICE_LOC', 
+         function($http, $sessionStorage, $q, PortalGroupService, SEARCH_URLS, filterFilter, miscService, SERVICE_LOC){
```
Comment (timlevett) [inline angularjs-portal-home/src/main/webapp/my-app/search/services.js]: `SERVICE_LOC` is being injected but not used

Comment (vertein) [inline angularjs-portal-home/src/main/webapp/my-app/search/services.js]: ha, good catch

File: angularjs-portal-home/src/main/webapp/my-app/search/services.js [hunk 3]
```diff
@@ -33,30 +33,103 @@ define(['angular', 'jquery'], function(angular, $) {
...
-              return response.data;
-            },
-            function(response){
-              console.log("error searching the wisc diretory: " +  response.status);
+        var directoryUrlPromise;
+        var directorySearchEnabledPromise;
+        
+        function directorySearch(term) {
+          return getDirectorySearchURL().then(function(searchDirectoryURL){
+            return $q(function(resolve, reject){
```
Comment (timlevett) [inline angularjs-portal-home/src/main/webapp/my-app/search/services.js]: neat

**Comment by timlevett** [comment]
:+1:

**Comment by keirserrie** [comment]
:+1:

**Comment by apetro** [comment]
:+1:

**Pull request merged**